villain/compile.rkt

```diff
@@ -73,6 +79,339 @@
       'raise_error
       'raise_error_align))
 
+;; Expr CEnv Boolean -> Asm
```

Comment:
I can't tell what's different here because the diff includes so much stuff.  What happened?

Reply:
I changed the place of the functions `(define (compile-e e c tail?) ...)`, `(define (compile-e-tail e c) ...)`, `(define (compile-e-nontail e c) ...)`, and `(define (move-args c-ct i) ...)` in the file, moving them up, but haven't changed these functions.  

The rest of the functions in this area (`(define (label-λ e) ...)`, `(define (fvs e) ...)`, `(define (match-p-fvs p) ...)`, `(define (λs e) ...)`, `(define (compile-λ-definitions ls) ...`, `(define (compile-λ-definition l) ...)`, etc.) are new.